api/src/main/java/org/apache/iceberg/actions/ActionsProvider.java

```diff
@@ -70,4 +70,10 @@ default RewritePositionDeleteFiles rewritePositionDeletes(Table table) {
     throw new UnsupportedOperationException(
         this.getClass().getName() + " does not implement rewritePositionDeletes");
   }
+
+  /** Instantiates an action to remove expired files. */
+  default RemoveExpiredFiles removeExpiredFiles(Table table) {
```

Comment:
@huaxingao: Can you please provide more background about why this interface is needed and what spark action we are planning to build on top of it? As per me, expireSnapshots action itself can clean the expired data and metadata files. So, I am wondering why this is required?

Reply:
This is part of earlier PR https://github.com/apache/iceberg/pull/4705. I think this PR was separated out to get consensus on the interface for this action.

Reply:
Yes, this PR was separated out from https://github.com/apache/iceberg/pull/4705 to get consensus on the interface for this action.

The `ExpireSnapshots` action does delete the files associated with expired snapshots. However, the `RemoveExpiredFiles` action serves a different purpose. While `ExpireSnapshots` removes the expired snapshots from the top-level metadata, some files that should be deleted might still remain due to actions like table copying. Essentially, `RemoveExpiredFiles` allows the target table to simulate the `ExpireSnapshots` behavior in the source table, ensuring that the files are cleaned up appropriately.

Reply:
a) Let me go through the `table copying` PR and the changes looks like it is specifically needed only for table copy.
b) Can't the existing remove orphan files clean those left over files after table copy?
c) `RemoveExpiredFiles`, I definitely want the naming to be different. It is already confusing. Maybe as `RemoveTableCopyOrphanFiles`

Reply:
Yes, the existing remove orphan files can clean those left over files, but it is very expensive, especially for S3.

Reply:
I think the name and purpose here are a little confusing for a public API. I would agree with @ajantha-bhat that we should consider keeping this a private part of the copy table action. If we are going to make it public maybe it should be a sub-function of that action, rather than top level api here.